app/experimenter/nimbus-ui/src/hooks/useChangeOperationMutation.test.tsx

```diff
@@ -20,28 +19,68 @@ import { useChangeOperationMutation } from "./useChangeOperationMutation";
 
 describe("hooks/useChangeOperationMutation", () => {
   it("can successfully execute mutation set callbacks", async () => {
-    const { callbacks, submitError } = setupTestHook();
+    const { result, invokeCallback, waitForNextUpdate } = setupTestHook();
 
-    for (const callback of callbacks) {
-      await act(async () => void callback());
-      await waitFor(() => expect(submitError).toBeNull());
+    const callbackCount = result.current.callbacks.length;
+    for (let idx = 0; idx < callbackCount; idx++) {
+      invokeCallback(idx);
+
+      await waitForNextUpdate();
+      expect(result.current.isLoading).toBeTruthy();
+
+      await waitForNextUpdate();
+      expect(result.current.isLoading).toBeFalsy();
+      expect(result.current.submitError).toBeNull();
     }
   });
 
-  it("indicates when loading", async () => {
-    const {
-      isLoading,
-      callbacks: [callback],
-    } = setupTestHook();
-    expect(isLoading).toBeFalsy();
-    await act(async () => void callback());
-    waitFor(() => expect(isLoading).toBeTruthy());
+  it("indicates when loading mutation", async () => {
+    const { result, invokeCallback, waitForNextUpdate } = setupTestHook();
+    expect(result.current.isLoading).toBeFalsy();
+    invokeCallback();
+
+    await waitForNextUpdate();
+    expect(result.current.isLoading).toBeTruthy();
+
+    await waitForNextUpdate();
+    expect(result.current.isLoading).toBeFalsy();
+  });
+
+  it("indicates when loading refetch", async () => {
+    const refetchState = {
+      called: false,
+      resolve: null as null | (() => void),
+    };
+    const refetch = () => {
+      refetchState.called = true;
+      return new Promise(
+        (resolve) => (refetchState.resolve = () => resolve(null)),
+      );
+    };
```

Comment:
This feels a little dirty, but seems like a decent way to predictably walk through the states around refetch.